packages/rocketchat-livechat/.app/client/views/messageAttachment.js

```diff
@@ -0,0 +1,25 @@
+import visitor from '../../imports/client/visitor';
+
+const fixLivechatUpload = function(url) {
+	let livechatUrl = url;
+
+	const query = `rc_room_type=${ 'l' }&rc_rid=${ visitor.getRoom() }&rc_token=${ visitor.getToken() }`;
+	if (livechatUrl.indexOf('?') === -1) {
+		livechatUrl = `${ livechatUrl }?${ query }`;
+	} else {
+		livechatUrl = `${ livechatUrl }&${ query }`;
+	}
+
+	return Meteor.absoluteUrl().replace(/\/$/, '') + livechatUrl;
+};
+
+Template.messageAttachment.helpers({
+	fixLivechatUpload,
```

Comment:
what about change this to a cookie approach just like [rocket.chat client does](https://github.com/RocketChat/Rocket.Chat/blob/develop/packages/rocketchat-file-upload/client/lib/fileUploadHandler.js#L32)?

Reply:
Fixed.